usecases/modules/compare.go

```diff
@@ -69,6 +72,9 @@ func reVectorizeMulti(ctx context.Context,
 	targetVector string,
 	findObjectFn modulecapabilities.FindObjectFn,
 ) (bool, models.AdditionalProperties, [][]float32, error) {
+	if ctx.Err() != nil {
+		return false, nil, nil, fmt.Errorf("context is done: %w", ctx.Err())
+	}
```

Comment:
you currently already have a context check in the loop that calls this function. Currently we have two context.Done checks per loop iteration